src/display/editor/drawers/inkdraw.js

```diff
@@ -0,0 +1,900 @@
+        const [c1x, c1y, c2x, c2y, x, y] = line
+          .subarray(i, i + 6)
+          .map(Outline.svgRound);
+        buffer.push(`C${c1x} ${c1y} ${c2x} ${c2y} ${x} ${y}`);
+      }
+    }
+    return buffer.join("");
+  }
+
+  serialize([pageX, pageY, pageWidth, pageHeight], isForCopying) {
+    const serializedLines = [];
+    const serializedPoints = [];
+    let tx, ty, sx, sy, swap;
+    const [x, y, width, height] = this.#getBBoxWithNoMargin();
+    let x1, y1, x2, y2;
+
+    switch (this.#rotation) {
+      case 0:
+        swap = false;
+        tx = pageX;
+        ty = pageY + pageHeight;
+        sx = pageWidth;
+        sy = -pageHeight;
+        x1 = pageX + x * pageWidth;
+        y1 = pageY + (1 - y - height) * pageHeight;
+        x2 = pageX + (x + width) * pageWidth;
+        y2 = pageY + (1 - y) * pageHeight;
+        break;
+      case 90:
+        swap = true;
+        tx = pageX;
+        ty = pageY;
+        sx = pageWidth;
+        sy = pageHeight;
+        x1 = pageX + y * pageWidth;
+        y1 = pageY + x * pageHeight;
+        x2 = pageX + (y + height) * pageWidth;
+        y2 = pageY + (x + width) * pageHeight;
+        break;
+      case 180:
+        swap = false;
+        tx = pageX + pageWidth;
+        ty = pageY;
+        sx = -pageWidth;
+        sy = pageHeight;
+        x1 = pageX + (1 - x - width) * pageWidth;
+        y1 = pageY + y * pageHeight;
+        x2 = pageX + (1 - x) * pageWidth;
+        y2 = pageY + (y + height) * pageHeight;
+        break;
+      case 270:
+        swap = true;
+        tx = pageX + pageWidth;
+        ty = pageY + pageHeight;
+        sx = -pageWidth;
+        sy = -pageHeight;
+        x1 = pageX + (1 - y - height) * pageWidth;
+        y1 = pageY + (1 - x - width) * pageHeight;
+        x2 = pageX + (1 - y) * pageWidth;
+        y2 = pageY + (1 - x) * pageHeight;
+        break;
+    }
+
+    if (swap) {
+      for (const { line, points } of this.#lines) {
+        serializedLines.push(
+          Outline._rescaleAndSwap(
+            line,
+            tx,
+            ty,
+            sx,
+            sy,
+            isForCopying ? new Array(line.length) : null
+          )
+        );
+        serializedPoints.push(
+          Outline._rescaleAndSwap(
+            points,
+            tx,
+            ty,
+            sx,
+            sy,
+            isForCopying ? new Array(points.length) : null
+          )
+        );
+      }
+    } else {
+      for (const { line, points } of this.#lines) {
+        serializedLines.push(
+          Outline._rescale(
+            line,
+            tx,
+            ty,
+            sx,
+            sy,
+            isForCopying ? new Array(line.length) : null
+          )
+        );
+        serializedPoints.push(
+          Outline._rescale(
+            points,
+            tx,
+            ty,
+            sx,
+            sy,
+            isForCopying ? new Array(points.length) : null
+          )
+        );
+      }
```

Comment:
Unless I'm missing something this code *appears* to be essentially duplicated, can that perhaps be avoided?
```suggestion
    const rescaleFn = swap ? Outline._rescaleAndSwap : Outline._rescale;
    for (const { line, points } of this.#lines) {
      serializedLines.push(
        rescaleFn(
          line,
          tx,
          ty,
          sx,
          sy,
          isForCopying ? new Array(line.length) : null
        )
      );
      serializedPoints.push(
        rescaleFn(
          points,
          tx,
          ty,
          sx,
          sy,
          isForCopying ? new Array(points.length) : null
        )
      );
    }
```